### DISCUSSION TITLE
Variable shape

### DISCUSSION THREAD

**Discussion by Anmol-Bakshi** [neperfepx/neper#882] [Q&A] [answered]
How can i make this shape shown in image containing "n" number of grains?
![image](https://github.com/neperfepx/neper/assets/108575045/f38c8575-b394-4510-be97-3fb5432d7012)

**Comment by Srijan7Y** [comment]
Apply the geometry and use _-planecut_ or _-hspace_ under _-transform_ , accordingly for the same. Refer to **#862** , I have mentioned the commands there.
Though I am not sure about the number of grains as I got 43 grains.

**Reply by Anmol-Bakshi** [reply]
i tried but i am getting bug, how to resolve.
neper -T -n 100 -dim 2 -morpho gg -morphooptistop "val=1e-1" -domain "square(8,3.1)" -transform "cut(hspace(0,1,0,-0), hspace(0,-1,0,3.1), hspace(1,0,0,-0), hspace(-1,0,0,8), hspace(1,-2.433,0,11.897))" -id 2 -o trapezoid_2d

========================    N   e   p   e   r    =======================
Info   : A software package for polycrystal generation and meshing.
Info   : Version 4.9.1-1
Info   : Built with: gsl|muparser|opengjk|openmp|nlopt|libscotch (full)
Info   : Running on 48 threads.
Info   : <https://neper.info>
Info   : Copyright (C) 2003-2022, and GNU GPL'd, by Romain Quey.
Info   : No initialization file found (`/home/ffg/.neperrc').
Info   : ---------------------------------------------------------------
Info   : MODULE  -T loaded with arguments:
Info   : [ini file] (none)
Info   : [com line] -n 100 -dim 2 -morpho gg -morphooptistop val=1e-1
         -domain square(8,3.1) -transform cut(hspace(0,1,0,-0),
         hspace(0,-1,0,3.1), hspace(1,0,0,-0), hspace(-1,0,0,8),
         hspace(1,-2.433,0,11.897)) -id 2 -o trapezoid_2d
Info   : ---------------------------------------------------------------
Info   : Reading input data...
Info   : Creating domain...
Info   : Creating tessellation...
Info   :   - Setting seeds... 100%
Info   :   - Running tessellation...
Info   :     > Initial solution: f   =0.670936689
Info   :     > Iteration    933: fmin=0.099964939 f=0.099964939
Info   :     > Final solution  : f   =0.099964939 (933 iterations)
Info   :     > Reached `val' criterion.
Info   : Generating crystal orientations...
Info   : Transforming tessellation...
Info   :   - Cutting (experimental)...
Info   :     > hspace(0,1,0,-0)...
Info   :     >  hspace(0,-1,0,3.1)...
Info   :     >  hspace(1,0,0,-0)...
Info   :     >  hspace(-1,0,0,8)...
Info   :     >  hspace(1,-2.433,0,11.897)...
Error  : You have discovered a bug in Neper!  Please file an issue at
         https://github.com/rquey/neper/issues.  Thank you.

Aborted (core dumped)

**Reply by Srijan7Y** [reply]
Try changing the val criteria maybe, I myself was not able to generate a 2d shape for this.  
Generally such issues are related to the geometrical dimensions given, check them thoroughly, there might be some error.

**Comment by rquey** [comment, marked as answer]
This discussion has been closed as obsolete. Please repost if necessary.
